### PR TITLE
[grpc] update to 1.44.0

### LINKED ISSUE #23521: [grpc] update to 1.44.0
Library name: grpc

New version number: 1.44.0

https://github.com/grpc/grpc/releases

### PR DESCRIPTION
**Pull request by FrankXie05**
Fix #23521

Update [grpc] to the latest version 1.44.0

All features have been tested successfullly in the following triplet:

- x64-windows
- x86-windows
- x64-windows-static

### CODE REVIEW

**Review by JonLiu1993 — APPROVED** [review]

File: ports/grpc/00003-undef-base64-macro.patch [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
-@@ -97,6 +97,8 @@ void grpc_stream_ref_init(grpc_stream_refcount* refcount, int /*initial_refs*/,
-       refcount, &refcount->slice_refcount);
+@@ -77,6 +77,8 @@ void grpc_stream_ref_init(grpc_stream_refcount* refcount, int /*initial_refs*/,
+                                                              : nullptr);
  }
  
 +#undef move64
 +
- static void move64(uint64_t* from, uint64_t* to) {
+ static void move64bits(uint64_t* from, uint64_t* to) {
```
Comment (joschiwald) [inline ports/grpc/00003-undef-base64-macro.patch:11]: is this patch is still required after the function has been renamed? https://github.com/grpc/grpc/commit/d88a57b35088abb2afc2f7a6074c27d21ccced0e

File: ports/grpc/00004-link-gdi32-on-windows.patch [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 +++ b/CMakeLists.txt
-@@ -278,7 +278,7 @@ elseif(UNIX)
- endif()
+@@ -323,7 +323,7 @@ include(cmake/xxhash.cmake)
+ include(cmake/zlib.cmake)
  
  if(WIN32)
--  set(_gRPC_BASELIB_LIBRARIES wsock32 ws2_32 crypt32)
+-  set(_gRPC_BASELIB_LIBRARIES ws2_32 crypt32)
 +  set(_gRPC_BASELIB_LIBRARIES wsock32 ws2_32 crypt32 gdi32)
```
Comment (joschiwald) [inline ports/grpc/00004-link-gdi32-on-windows.patch:10]: this patch should only add link against gdi32, but now it add link against wsock32 which has been remove in grpc 1.42.0 https://github.com/grpc/grpc/commit/05e5b085a56e7b4a6f07a785d7c91c3c2d8e2bfa

Comment (FrankXie05) [reply ports/grpc/00004-link-gdi32-on-windows.patch:10]: @joschiwald Thanks for your review, I saw that grpc was updated to version 1.45 the day before yesterday, I will update it again and remove the patch that has been fixed. :)

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**